.github/workflows/e2e-test.yml

```
@@ -6,85 +6,38 @@ on:
       - master
 
 jobs:
-  ubuntu-16:
-    runs-on: ubuntu-16.04
-    name: 'Ubuntu 16.04'
+  e2e:
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        password: [root]
+        node: [12.x]
     steps:
-      - run: sudo systemctl start mysql.service && sudo apt-get update && sudo apt-get install -y nginx
-        name: 'Setting up required services'
-      - run: 'echo -e "127.0.0.1 cli-testing.ghost.org\n" | sudo tee -a /etc/hosts'
-        name: 'Updating hosts file'
+      - name: Installing nginx
+        run: sudo apt-get update && sudo apt-get install -y nginx
+      - name: Updating hosts file
+        run: 'echo -e "127.0.0.1 cli-testing.ghost.org\n" | sudo tee -a /etc/hosts'
       - uses: actions/checkout@v2
         with:
           path: 'cli'
       - uses: actions/setup-node@v1
         with:
-          node-version: '12.x'
-      - run: cd cli && yarn && sudo yarn link --link-folder /usr/bin/
-      - name: Setting up Ghost instance
-        run: |
-          ghost install \
-            -d ghost \
-            --auto \
-            --no-prompt \
-            --url http://cli-testing.ghost.org \
-            --db mysql \
-            --dbhost localhost \
-            --dbuser root \
-            --dbpass root \
-            --dbname ghost-ubuntu16
-      - name: Verifying Installation
-        run: curl http://cli-testing.ghost.org | grep ghost
-
-  ubuntu-18:
-    runs-on: ubuntu-18.04
-    name: 'Ubuntu 18.04'
-    steps:
-      - run: sudo systemctl start mysql.service && sudo apt-get update && sudo apt-get install -y nginx
-        name: 'Setting up required services'
-      - run: 'echo -e "127.0.0.1 cli-testing.ghost.org\n" | sudo tee -a /etc/hosts'
-        name: 'Updating hosts file'
-      - uses: actions/checkout@v2
-        with:
-          path: 'cli'
-      - uses: actions/setup-node@v1
-        with:
-          node-version: '12.x'
-      - run: cd cli && yarn && sudo yarn link --link-folder /usr/bin/
-      - name: Setting up Ghost instance
+          node-version: ${{ matrix.node }}
+      - name: Start MySQL service
+        run: sudo systemctl start mysql.service
+      - name: Fix MySQL Password
+        if: ${{ matrix.os == 'ubuntu-20.04' }}
```

Github has a web IDE for editing workflow files, boasting a decent level of maturity. It has support for auto complete of keys in many places which is really cool. I got burned a few times because of variable interpolation errors, though.

---

So one comment here is that `if` keys don't need variable interpolation, so you _should_ be able to do 
```suggestion
        if: matrix.os == "ubuntu-20.04"
```